src/Algebra/Graph.hs

```diff
@@ -1041,3 +1054,48 @@ sparsify graph = res
         m <- get
         put (m + 1)
         overlay <$> s `x` m <*> m `y` t
+
+buildG :: forall a. F.Fold a -> Graph a
+buildG g = F.foldg Empty Vertex Overlay Connect g
+{-# INLINE [1] buildG #-}
+
+-- | FB functions
+mapGFB :: (b -> c) -> (a -> b) -> a -> c
+mapGFB = (.)
+{-# INLINE [0] mapGFB #-}
+
+induceFB :: b -> (a -> b) -> (a -> Bool) -> a -> b
+induceFB e v p = \x -> if p x then v x else e
+{-# INLINE [0] induceFB #-}
+
+transposeFB :: (b -> b -> b) -> (b -> b -> b)
```

Comment:
Wait, I thought we no longer need anything special for `transpose`, because it is written as a `foldg`. Instead, we could add rules that rewrite `foldg e _ _ _ Empty` into `e`, etc.

Reply:
W need such function to "rewrite back" un-merged expression :)

Concerning the `foldg` rules, this is not so simple due to the `transpose . star` problem...

Reply:
My response ended up being too long, so I posted it in the comments below.